### PR TITLE
Add explanation for API design

### PR DESCRIPTION
**Pull request by SonderSwift**
LGTM means you think this commit is good.

### CODE REVIEW

**Review by stannrodriguez — APPROVED** [review]
Great explanations

File: README.md [hunk 1]
```diff
@@ -19,4 +19,148 @@ git commit -m "commit message here"
...
+
+### General Components of an Application
+1. Frontend (Web GUI)
+	- (1990s/2000s) HTML, CSS server-side rendering
+		- Go to an internet endpoint specified by ip address/domain name
+		- directory
+			- index.html
+				- references other files
+				- each file is a new page load
+				- within each file may be a CSS tag and/or JS tag which helps the browser render styles and funcionality
```
Comment (SonderSwift) [inline README.md:32]: ooooo

File: README.md [hunk 2]
```diff
@@ -19,4 +19,148 @@ git commit -m "commit message here"
...
+		- be able to create account/sign-in/auth
+	- be able to compare dogs
+	- be able to donate money to a single dog or many dogs
+	- be able to buy food treats for dogs
+	- be able to buy clothes for dogs
+	- be able to buy toys for dogs
+	- view assortment of treats, clothes, and toys (on separate pages)
+	- be able to adopt a dog
+* DB of:
+	- dogs
```
Comment (rodriguez-jan) [inline README.md:87]: what about cats

File: README.md [hunk 3]
```diff
@@ -19,4 +19,148 @@ git commit -m "commit message here"
...
+	- get details on a single dog
+	- be able to verify user is real person
+		- be able to create account/sign-in/auth
+	- be able to compare dogs
+	- be able to donate money to a single dog or many dogs
+	- be able to buy food treats for dogs
+	- be able to buy clothes for dogs
+	- be able to buy toys for dogs
+	- view assortment of treats, clothes, and toys (on separate pages)
+	- be able to adopt a dog
```
Comment (rodriguez-erielle) [inline README.md:85]: 🐶

**Pull request merged**